packages/manager/src/features/CloudPulse/Utils/FilterBuilder.ts

```diff
@@ -219,23 +250,23 @@ export const buildXFilter = (
   }
 ): Filter => {
   const filters: Filter[] = [];
+  let orCondition: Filter[] = [];
 
   const { dependency } = config.configuration;
   if (dependency) {
     dependency.forEach((key) => {
       const value = dependentFilters[key];
       if (value !== undefined) {
         if (Array.isArray(value)) {
-          const orCondition = value.map((val) => ({ [key]: val }));
-          filters.push({ '+or': orCondition });
```

Comment:
we discussed the existing logic will work right, why we need it now?

Reply:
actually, while re-checking i found out that existing logic will not work, because existing logic doesnt filter correctly based on tags, for ex- if a resource doesnt have the mentioned tag, it will barely filter based on region and give all the resources based on region